x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/hooks.tsx

```diff
@@ -232,7 +233,17 @@ export const useActionsLogFilter = ({
         }))
       : isHostsFilter
       ? []
-      : RESPONSE_ACTION_API_COMMANDS_NAMES.map((commandName) => ({
+      : RESPONSE_ACTION_API_COMMANDS_NAMES.filter((commandName) => {
+          // `get-file` is currently behind FF
+          if (
+            commandName === 'get-file' &&
+            !ExperimentalFeaturesService.get().responseActionGetFileEnabled
+          ) {
+            return false;
+          }
+
+          return true;
+        }).map((commandName) => ({
```

Comment:
Consider replacing the `filter` and `map` with `reduce` so we don't iterate through the list twice. Something like this maybe

```suggestion
      : RESPONSE_ACTION_API_COMMANDS_NAMES.reduce<FilterItems>((acc, cmd) => {
          // `get-file` is currently behind FF
          const commandName = !ExperimentalFeaturesService.get().responseActionGetFileEnabled
            ? cmd === 'get-file'
              ? undefined
              : cmd
            : cmd;

          if (commandName) {
            acc.push({
              key: commandName,
              label: getUiCommand(commandName),
              checked:
                !isFlyout &&
                commands?.map((command) => getCommandKey(command)).includes(commandName)
                  ? 'on'
                  : undefined,
              'data-test-subj': `${filterName}-filter-option`,
            });
          }

          return acc;
        }, [])
```

Reply:
I'd object. I don't think we'd win a lot on performance in practice - theoretically we'd win nothing as O(2*n) = O(n).

On the other hand, the separate `filter` and `map` are easier on the eye (for me at least), and it's easier to add/remove the temporary `filter` step without modifying existing logic.

Reply:
@ashokaditya ,
I actually had tried a different approach that avoided the double loop, but started to get TS types miss-matches, so since this is temporary and as @gergoabraham mentioned, having the additional `.filter()` will make it cleaner to remove once we re-enable `get-file`

Reply:
> I'd object. I don't think we'd win a lot on performance in practice - theoretically we'd win nothing as O(2*n) = O(n).

Sure, if you're considering algorithmic complexity then `n` and `2n` are both linear time in terms of big O. However, I am looking at this practically, where doing something in n steps is still better than doing it in 2n steps. For a sufficiently large list, I would not recommend scanning it twice when the same can be done with a single scan. Here in the response actions list though, there are just 6 items so it doesn't matter if we `filter` and then `map`.

> On the other hand, the separate `filter` and `map` are easier on the eye (for me at least), and it's easier to add/remove the temporary `filter` step without modifying existing logic.

Agree. 👍